### PR TITLE
stirde_properties fix

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes part of #6015

profiled permutation order is wrong and nvfuser generates output in wrong memory format. Though this problem doesn't seem to cause any issue with the test (except a graceful fallback path taken by CudaFusionGuard).

### CODE REVIEW

**Comment by jjsjann123** [comment]
cc'ing @ngimel since this is the third (AFAIK) patch to compute stride properties. :pleading_face:

File: aten/src/ATen/test/stride_properties_test.cpp [hunk 1]
```diff
@@ -69,10 +69,25 @@ TEST(StridePropertiesTest, ZeroStrideIndicesEagerConsistencyTest) {
...
+
+  auto temp = TensorType::create(c10::nullopt, c10::nullopt, t.sizes(), t.strides(), c10::nullopt);
+  TORCH_INTERNAL_ASSERT(temp->stride_properties().isComplete() &&
+      temp->stride_properties().isComplete(), "complete stride properties is needed for the test");
+  std::vector<size_t> stride_indices(2);
+  std::iota(stride_indices.rbegin(), stride_indices.rend(), 0);
+
+  auto index_iter = stride_indices.begin();
+  for (const auto& opt_stride : *temp->stride_properties().sizes()) {
+    EXPECT_TRUE(*index_iter++ != opt_stride->stride_index_.value());
```
Comment (davidberard98) [inline aten/src/ATen/test/stride_properties_test.cpp:91]: this is failing ... which I think makes sense @jjsjann123

Comment (jjsjann123) [reply aten/src/ATen/test/stride_properties_test.cpp:91]: errr... I was dumb.... That's a copy-paste went wrong....

**Review by davidberard98 — APPROVED** [review]
LGTM

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**